Assignment operators in til::spsc don't return anything

# Environment

Windows build number: Version 10.0.18363.1198
Windows Terminal version (if applicable): 

# Steps to reproduce

I was trying to use the `til::spsc` class to do something like this:

```
auto channel = til::spsc::channel<QueueItem>(100);
auto producer = std::move(channel.first);
channel.first = std::move(producer);
```

# Expected behavior

The code should compile.

# Actual behavior

It fails with the error:
```
C:\terminal\src\inc\til\spsc.h(429): error C4716: 'til::spsc::producer<QueueItem>::operator=': must return a value
```

I believe the problem is that the assignment operators in the `producer` and `consumer` classes don't actually return anything. 

See here:
https://github.com/microsoft/terminal/blob/ae550e0969595f062b15c2ff5cc33d4afe8ebc3f/src/inc/til/spsc.h#L424-L428

and here:
https://github.com/microsoft/terminal/blob/ae550e0969595f062b15c2ff5cc33d4afe8ebc3f/src/inc/til/spsc.h#L542-L546

I'm assuming they should have a `return *this;` or something along those lines.

--- a/src/inc/til/spsc.h
+++ b/src/inc/til/spsc.h
@@ -425,6 +425,7 @@ namespace til::spsc
         {
             drop();
             _arc = std::exchange(other._arc, nullptr);
+            return *this;
         }
 
         ~producer()
@@ -543,6 +544,7 @@ namespace til::spsc
         {
             drop();
             _arc = std::exchange(other._arc, nullptr);
+            return *this;
         }
 
         ~consumer()

--- a/src/til/ut_til/SPSCTests.cpp
+++ b/src/til/ut_til/SPSCTests.cpp
@@ -50,11 +50,43 @@ class SPSCTests
         TEST_CLASS_PROPERTY(L"TestTimeout", L"0:0:10") // 10s timeout
     END_TEST_CLASS()
 
+    TEST_METHOD(SmokeTest);
     TEST_METHOD(DropEmptyTest);
     TEST_METHOD(DropSameRevolutionTest);
     TEST_METHOD(DropDifferentRevolutionTest);
     TEST_METHOD(IntegrationTest);
 };
+
+void SPSCTests::SmokeTest()
+{
+    // This test mostly ensures that the API wasn't broken.
+
+    // construction
+    auto [tx, rx] = til::spsc::channel<int>(32);
+    std::array<int, 3> data{};
+
+    // move constructor
+    til::spsc::producer tx2(std::move(tx));
+    til::spsc::consumer rx2(std::move(rx));
+
+    // move assignment operator
+    tx = std::move(tx2);
+    rx = std::move(rx2);
+
+    // push
+    tx.emplace(0);
+    tx.push(data.begin(), data.end());
+    tx.push(til::spsc::block_initially, data.begin(), data.end());
+    tx.push(til::spsc::block_forever, data.begin(), data.end());
+    tx.push_n(data.begin(), data.size());
+    tx.push_n(til::spsc::block_initially, data.begin(), data.size());
+    tx.push_n(til::spsc::block_forever, data.begin(), data.size());
+
+    // pop
+    std::optional<int> x = rx.pop();
+    rx.pop_n(til::spsc::block_initially, data.begin(), data.size());
+    rx.pop_n(til::spsc::block_forever, data.begin(), data.size());
+}
 
 void SPSCTests::DropEmptyTest()
 {

